src/cli/add.rs

```diff
@@ -160,14 +171,47 @@ pub async fn add_specs_to_project(
     }
 
     // Update the lock file and write to disk
-    update_lock_file(
+    let lock_file = update_lock_file(
         project,
         load_lock_file(project).await?,
         Some(sparse_repo_data),
     )
     .await?;
     project.save()?;
 
+    if !no_install {
+        let platform = Platform::current();
```

Comment:
Why can't `get_up_to_date_prefix` be used here?

Reply:
Because `get_up_to_date_prefix` also re-reads the lock file and updates it, all of that has already happened in the code above so its a bit of a waste. The code that is in this if statement is basically roughly the same as the last part of that function.